src/common/VirtualObject.cpp

```diff
@@ -25,6 +26,36 @@ IObject *VirtualObject::clone() const {
     for (const auto *const component: _components) {
         newObject->addComponent(component->clone(newObject));
     }
+    int posa = 0;
+    for (auto *const component: _components) {
+        int posb = 0;
+        for (const auto *const fieldGroup: component->getMeta().getFieldGroups()) {
+            int posc = 0;
+            for (auto *const field: fieldGroup->getFields()) {
+                if (dynamic_cast<ComponentField *>(field) != nullptr) {
+                    const UUID uuid = dynamic_cast<ComponentField *>(field)->_uuid;
+                    int pos = 0;
+                    for (const auto *const comp: _components) {
+                        if (comp->getUUID() == uuid) {
+                            break;
+                        }
+                        pos++;
+                    }
+                    if (pos == _components.size()) {
+                        LOG.error("Component not found");
+                        continue;
+                    }
+                    auto *newComponent = newObject->_components[pos];
+                    auto *newField = newObject->_components[posa]->getMeta().getFieldGroups()[posb]->getFields()[posc];
```

Comment:
isn't possible to get the pointer to new group to avoid fetching it for each field ?
Like
`
int posb = 0;
IComponent *currentComp = newObject->_components[posa];
`
And the same for the groups and then the field.
Not mandatory but would be cleaner I believe.